### PR TITLE
SHTOOLS: Provide missing C wrapper for MakeGradientDH

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: S/SHTOOLS/build_tarballs.jl [hunk 1]
```diff
@@ -13,24 +13,10 @@ script = raw"""
 cd $WORKSPACE/srcdir/SHTOOLS-*
```
Comment (giordano) [inline S/SHTOOLS/build_tarballs.jl:13]: Need to add the source
```julia
DirectorySource("./bundled"),
```

Comment (eschnett) [reply S/SHTOOLS/build_tarballs.jl:14]: Done!

File: S/SHTOOLS/bundled/patches/add-cMakeGradientDH.patch [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
++        real(kind=c_double), dimension(theta_d0,theta_d1),intent(out) :: phi
++        integer(kind=c_int), value,intent(in) :: lmax
++        integer(kind=c_int), intent(out) :: n
++        integer(kind=c_int), optional,intent(in) :: sampling
++        integer(kind=c_int), optional,intent(in) :: lmax_calc
++        integer(kind=c_int), optional,intent(in) :: extend
++        integer(kind=c_int), optional,intent(out) :: exitstatus
++        call MakeGradientDH(cilm,lmax,theta,phi,n,sampling=sampling&
++                                ,lmax_calc=lmax_calc,extend=extend,exitstatus=e
+xitstatus)
```
Comment (giordano) [inline S/SHTOOLS/bundled/patches/add-cMakeGradientDH.patch:28]: Probably
```suggestion
+                                ,lmax_calc=lmax_calc,extend=extend,exitstatus=exitstatus)
```
?

Comment (eschnett) [reply S/SHTOOLS/bundled/patches/add-cMakeGradientDH.patch]: Accident...

**Pull request merged**